src/main/java/io/reactivex/Single.java

```diff
@@ -2302,6 +2302,44 @@
         return delaySubscription(Observable.timer(time, unit, scheduler));
     }
 
+    /**
+     * Maps the {@link Notification} success value of this Single back into normal
+     * {@code onSuccess}, {@code onError} or {@code onComplete} signals as a
+     * {@link Maybe} source.
+     * <p>
+     * Note that {@code this} should be of type {@code Single<Notification<T>>} or
+     * the transformation will result in an {@code onError} signal of
+     * {@link ClassCastException}. Currently, the Java language doesn't allow specifying
+     * methods for certain type argument shapes only (unlike extension methods would),
+     * hence the forced casting in this operator.
+     * <p>
+     * In addition, usually the inner value type (T2) has to be expressed again via
+     * a type argument on this method (see example below).
+     * <dl>
+     * <dt><b>Scheduler:</b></dt>
+     * <dd>{@code dematerialize} does by default subscribe to the current Single
+     * on the {@link Scheduler} you provided, after the delay.</dd>
+     * </dl>
+     * <p>
+     * Example:
+     * <pre><code>
+     * Single.just(Notification.createOnNext(1))
+     * .&lt;Integer&gt;dematerialize()
+     * .test()
+     * .assertResult(1);
+     * </code></pre>
+     * @param <T2> the type inside the Notification
+     * @return the new Maybe instance
+     * @since 2.2.4 - experimental
+     */
+    @SuppressWarnings("unchecked")
+    @CheckReturnValue
+    @SchedulerSupport(SchedulerSupport.NONE)
+    @Experimental
+    public final <T2> Maybe<T2> dematerialize() {
```

Comment:
I find this API very awkward since it's available on every type but should only be used on `Single<Notification<T>>`. I don't have a better suggestion though. Except defining it yourself and using a Kotlin extension function.

Reply:
Didn't know about `dematerialize`, then I guess we stick to that convention. Providing a mapper sounds good. Then maybe we could deprecate the current `dematerialize` in Observable.

Reply:
Okay then. I'll change this to use the selector and post a separate PR for adding a selector variant to `Observable`/`Flowable` and deprecate the type-fragile versions.